pipeline/sns.py

```diff
@@ -203,26 +210,68 @@ def sns_alert_system(earthquakes: list[dict]):
     conn = get_connection()
 
     topics = get_topics(conn)
-    related_topics = find_related_topics(earthquakes, topics)
-    subscribed_users = get_subscribed_users(conn, related_topics)
-
-    for user in subscribed_users:
-        magnitude = user['min_magnitude']
-        message = (
-            f"Earthquake Alert! Magnitude {magnitude} or above "
-            "earthquake detected in your area"
-        )
-        subject = "Earthquake Alert"
-        send_message(
-            sns_client, user['topic_arn'], subject, message
-        )
+    for earthquake in earthquakes:
+        related_topics = find_related_topics(earthquake, topics)
+        subscribed_users = get_subscribed_users(conn, related_topics)
+        for user in subscribed_users:
+            magnitude = earthquake['magnitude']
+            eq_coordinates = (earthquake['lon'], earthquake['lat'])
+            topic_coordinates = (user['lon'], user['lat'])
+            distance = calculate_distance(topic_coordinates, eq_coordinates)
+            message = (
+f"""
+
+**EARTHQUAKE WARNING!**
+
+A magnitude {magnitude} or above earthquake has been detected within {distance}km of your area.
+
+SAFETY TIPS:
+
+1. DROP, COVER, and HOLD ON: Drop to your hands and knees. Take cover under a sturdy table or desk, or cover your head and neck with your arms. Hold on until the shaking stops.
+
+2. STAY INDOORS: Remain inside; do not run outside while the building is shaking.
+
+3. AVOID WINDOWS AND HEAVY OBJECTS: Move away from windows, glass, and any objects that could fall, like bookcases or mirrors.
+
+4. IF OUTSIDE, FIND AN OPEN AREA: Move to a clear area away from buildings, trees, streetlights, and wires.
+
+5. IF IN A CAR, PULL OVER SAFELY: Pull over to a safe location and stay inside until the shaking stops.
+
+Stay safe and follow these guidelines until the shaking subsides.
```

Comment:
I like the detail of this message, but some of the wording could be a little clearer. Read through this and double check that it all makes sense.

Reply:
I believe it makes sense